packages/eas-cli/src/build/build.ts

```diff
@@ -649,3 +654,59 @@ function formatAccountSubscriptionsUrl(accountName: string): string {
     getExpoWebsiteBaseUrl()
   ).toString();
 }
+
+async function createAndUploadFingerprintAsync<T extends Platform>(
+  ctx: BuildContext<T>
+): Promise<{
+  runtimeVersion?: string;
+  fingerprintGCSLocation?: string;
+}> {
+  const resolvedRuntimeVersion = await resolveRuntimeVersionAsync({
+    exp: ctx.exp,
+    platform: ctx.platform,
+    workflow: ctx.workflow,
+    projectDir: ctx.projectDir,
+    env: ctx.buildProfile.env,
+    cwd: ctx.projectDir,
+  });
+
+  if (!resolvedRuntimeVersion?.fingerprintSources) {
+    return {
+      runtimeVersion: resolvedRuntimeVersion?.runtimeVersion ?? undefined,
+    };
+  }
+
+  const fingerprintLocation = path.join(getTmpDirectory(), `${uuidv4()}-runtime-fingerprint.json`);
+
+  await fs.writeJSON(fingerprintLocation, {
+    hash: resolvedRuntimeVersion.runtimeVersion,
+    sources: resolvedRuntimeVersion.fingerprintSources,
+  });
+
+  let fingerprintGCSLocation = undefined;
+  try {
+    fingerprintGCSLocation = await uploadFileAtPathToGCSAsync(
+      ctx.graphqlClient,
+      UploadSessionType.EasUpdateFingerprint,
+      fingerprintLocation
+    );
+  } catch (err: any) {
+    let errMessage = 'Failed to upload fingerprint to EAS';
+
+    if (err.message) {
+      errMessage += `\n\nReason: ${err.message}`;
+    }
+
+    Log.warn(errMessage);
+    return {
+      runtimeVersion: resolvedRuntimeVersion?.runtimeVersion ?? undefined,
+    };
```

Comment:
I think it makes sense to fail the build command if this isn't able to be uploaded?

Reply:
I think we shouldn't since it's not necessary for the build to work. We [don't fail the build](https://github.com/expo/eas-cli/blob/main/packages/eas-cli/src/build/build.ts#L336) if uploading metadata fails either.